Framework/Muon/inc/MantidMuon/ApplyMuonDetectorGroupPairing.h

```diff
@@ -0,0 +1,120 @@
+#ifndef MANTID_MUON_APPLYMUONDETECTORGROUPPAIRING_H_
+#define MANTID_MUON_APPLYMUONDETECTORGROUPPAIRING_H_
+
+#include "MantidAPI/Algorithm.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidKernel/System.h"
+#include "MantidMuon/MuonAlgorithmHelper.h"
+
+namespace Mantid {
+namespace Muon {
+
+/** ApplyMuonDetectorGroupPairing :
+
+@date 2018-06-01
+
+Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+
+class DLLExport ApplyMuonDetectorGroupPairing : public API::Algorithm {
+public:
+  /// (Empty) Constructor
+  ApplyMuonDetectorGroupPairing() : API::Algorithm() {}
+  /// Virtual destructor
+  ~ApplyMuonDetectorGroupPairing() {}
+  /// Algorithm's name
+  const std::string name() const override {
+    return "ApplyMuonDetectorGroupPairing";
+  }
+  /// Algorithm's version
+  int version() const override { return (1); }
+  /// Algorithm's category for identification
+  const std::string category() const override {
+    return "DataHandling\\Grouping;Transforms\\Grouping";
```

Comment:
we should put in muon too

Reply:
I will place all three algorithms in `Muon/DataHandling`